cypress/e2e/focusTrap.cy.js

```diff
@@ -190,4 +190,111 @@ describe('Focus trap', () => {
       }),
     })
   })
+
+  it('should ignore focusable shadow hosts if they delegate focus to their shadow subtree', () => {
+    cy.runExample({
+      html: stripIndent(/* html */ `
+      <div id="shadow-host-delegates-focus">
+        <p>
+          Custom Elements that delegate focus
+          should <i>not</i> return the host as a focusable element; only the
+          focusable elements in the shadow DOM.
+        </p>
+        <fancy-button data-cy-delegates-focus tabindex="0">I'm in the Shadow DOM</fancy-button>
+      </div>
+    `),
+      test: serialize(() => {
+        // Get the first focusable element, according to our library
+        cy.get('#shadow-host-delegates-focus')
+          .as('container')
+          .aliasFocusableEdges()
+
+        // Get the shadow host in this container
+        cy.get('@container').find('[data-cy-delegates-focus]').as('shadowHost')
+
+        // Assert that we have a shadow host that delegates focus to its subtree
+        cy.get('@shadowHost')
+          .shadow()
+          .should('have.prop', 'delegatesFocus', true)
+
+        // Assert that the shadowHost is *not* what our library returns
+        cy.get('@first')
+          .should('not.deep.equal', '@shadowHost')
```

Comment:
This does not work the way you expect it to be. This compares a DOM element to a string. You want:

```js
cy.get('@first').then(first => {
  cy.get('@shadowHost').should('not.deep.equal', first.get(0))
})